### ISSUE TITLE
An error should be raised if a non-existent kwarg is given 

### ISSUE THREAD

**Issue by loostrum** [dianna-ai/dianna#130]
Due to the use of `get_kwargs_applicable_to_function`, the use may provide a kwarg that doesn't exist, but is silently ignored as it is simply not returned by `get_kwargs_applicable_to_function`. 
If possible, it would be nice to check if all input kwargs are present in the union of all outputs of `get_kwargs_applicable_to_function` so we can raise an error when an unknown argument is passed.

**Comment by egpbos** [comment]
I think it's not uncommon for non-existent `kwargs` to go silently unnoticed. The major downside to raising errors is that it will be annoying boilerplate. However, it could probably be useful, for instance when users misspell some keyword and they get confused about why it doesn't work as they expected.

To minimize boilerplate, we can make a function something like so:

```python
def check_for_unused_kwargs_keys(kwargs_keys, *functions):
    function_keys = set()
    for function in functions:
        function_keys.update(get_kwargs_applicable_to_function(function))
    unused_kwargs_keys = [key not in function_keys for key in kwargs_keys]
    if len(unused_kwargs_keys) > 0:
        raise Exception(f"Unused kwargs keys {unused_kwargs_keys} found!")
```

And use that at the top of functions that use `get_kwargs_applicable_to_function`.

**Comment by egpbos** [comment]
I implemented `kwandl.forward_transitive` there especially for this case, see https://github.com/egpbos/kwandl/blob/main/test_kwandl.py#L321 and CTRL-f for `forward_transitive` to see some examples of how this works.

Warning: the implementation is not easy to read, but if you want I can explain specific parts on a call.

**Comment by loostrum** [comment]
Nice, I knew kwandl had something for it :) Do you think it's stable enough to use in DIANNA? 
Or should we still go with some simple(r) workaround?

For the above issue we can also do away with the get_kwargs calls in lime.explain and use something like 
```
def lime.explain(arg1, arg2, some_kwarg=0, explain_instance_kwargs={})
    explainer.explain_instance(arg1, **explain_instance_kwargs)
```

**Comment by loostrum** [comment]
Fixed in #693
